File: app/src/main/java/com/harsha/data/model/MyPojo.java
package com.harsha.data.model;


import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class MyPojo implements Parcelable
{
    @SerializedName("banner")
    @Expose
    private Banner banner;

    @SerializedName("categories")
    @Expose
    private ArrayList<Categories> categories;

    @SerializedName("topPicks")
    @Expose
    private TopPicks topPicks;

    protected MyPojo(Parcel in) {
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<MyPojo> CREATOR = new Creator<MyPojo>() {
        @Override
        public MyPojo createFromParcel(Parcel in) {
            return new MyPojo(in);
        }

        @Override
        public MyPojo[] newArray(int size) {
            return new MyPojo[size];
        }
    };

    public Banner getBanner ()
    {
        return banner;
    }

    public void setBanner (Banner banner)
    {
        this.banner = banner;
    }

    public ArrayList<Categories> getCategories ()
    {
        return categories;
    }

    public void setCategories (ArrayList<Categories> categories)
    {
        this.categories = categories;
    }

    public TopPicks getTopPicks ()
    {
        return topPicks;
    }

    public void setTopPicks (TopPicks topPicks)
    {
        this.topPicks = topPicks;
    }

    @Override
    public String toString()
    {
        return "ClassPojo [banner = "+banner+", categories = "+categories+", topPicks = "+topPicks+"]";
    }
}

File: app/src/main/java/com/harsha/data/model/MyPojo.kt
package com.harsha.data.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class MyPojo() :Parcelable{

    @SerializedName("banner")
    @Expose
    private var banner: Banner? = null

    @SerializedName("categories")
    @Expose
    private var categories: ArrayList<Categories>? = null

    @SerializedName("topPicks")
    @Expose
    private var topPicks: TopPicks? = null

    constructor(parcel: Parcel) : this() {
        banner = parcel.readParcelable(Banner::class.java.classLoader)
        topPicks = parcel.readParcelable(TopPicks::class.java.classLoader)
    }

    fun getBanner(): Banner? {
        return banner
    }

    fun setBanner(banner: Banner?) {
        this.banner = banner
    }

    fun getCategories(): ArrayList<Categories>?  {
        return categories
    }

    fun setCategories(categories: ArrayList<Categories>? ) {
        this.categories = categories
    }

    fun getTopPicks(): TopPicks? {
        return topPicks
    }

    fun setTopPicks(topPicks: TopPicks?) {
        this.topPicks = topPicks
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeParcelable(banner, flags)
        parcel.writeParcelable(topPicks, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MyPojo> {
        override fun createFromParcel(parcel: Parcel): MyPojo {
            return MyPojo(parcel)
        }

        override fun newArray(size: Int): Array<MyPojo?> {
            return arrayOfNulls(size)
        }
    }
}